types/react-jss/index.d.ts

```diff
@@ -3,7 +3,7 @@
 // Definitions by: Sebastian Silbermann <https://github.com/eps1lon>
 //                 James Lawrence <https://github.com/jlaw90>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// TypeScript Version: 2.9
```

Comment:
@sandersn  Why is this opened and then merged within 1 hour? Could you elaborate why an older version would break all of the sudden? 

The state of this repo is borderline frustrating. CI breaks unrelated or times out, PRs are closed by a bot without any human interaction and now older version break without any explanation given? Any issue that tracks this breakage?

Reply:
@eps1lon All fair questions. The root cause is that PRs are tested against typescript@next, which changes daily.

However, we were actually only checking *PRs* against typescript@next until this week. Unchanged packages could be broken on new versions of Typescript and only the people who installed it would know. Now there is a daily build here: https://typescript.visualstudio.com/TypeScript/_build?definitionId=15&_a=summary
I'm trying to get the build to pass, making the smallest changes I can to get there. Once it passes, then unrelated CI breaks should not happen.

react-jss' problem was that our improved variance checking in conditional types caught some previously-ignored errors in the keyof types. If I changed the semantics, please feel free to raise the issue here or create a fresh PR.

Here's the issue tracking the current breaks: Microsoft/TypeScript#29898